app/frontend/src/utils/transformUtils.js

```diff
@@ -64,3 +65,130 @@ export function attachAttributesToLinks(formSchemaComponents) {
+ * @param {String} allowLateInterval A string of days,Weeks,months, that tells form to be allowed for late submission for a particular period
+ * @param {Object[]} repeatUntil An object of Moment JS date, that tells form to be finally close after repetition on particular day date
+ * @param {String} scheduleType A string one of Manual, ClosingDate OR Period
+ * @param {Object[]} closeDate An object of Moment JS date, Forms closing date
+ * @returns {Object[]} An object array of Available dates in given period
+ */
+export function getAvailableDates(
+  keepAliveFor=0,
+  keepAliveForInterval='days',
+  formStartDate,
+  term=null,
+  interval=null,
+  allowLateTerm=null,
+  allowLateInterval=null,
+  repeatUntil,
+  scheduleType,
+  closeDate=null
+) {
+
+  let substartDate = moment(formStartDate);
+  repeatUntil = moment(repeatUntil);
+  let calculatedsubcloseDate = getCalculatedCloseSubmissionDate(substartDate,keepAliveFor,keepAliveForInterval,allowLateTerm,allowLateInterval,term,interval,repeatUntil,scheduleType,closeDate);
+  let availableDates = [];
+  if(calculatedsubcloseDate && term && interval) {
+    while (substartDate.isBefore(calculatedsubcloseDate)) {
+      let newDate = substartDate.clone();
+      if(substartDate.isBefore(repeatUntil)){
+        availableDates.push(Object({
+          startDate:substartDate.format('YYYY-MM-DD HH:MM:SS'),
+          closeDate:newDate.add(keepAliveFor,keepAliveForInterval).format('YYYY-MM-DD HH:MM:SS'),
+          graceDate:newDate.add(allowLateTerm,allowLateInterval).format('YYYY-MM-DD HH:MM:SS')
+        }));
+      }
+      substartDate.add(term,interval);
+    }
+  }
+
+  if((term == null && interval == null) && (keepAliveFor && keepAliveForInterval)){
+    let newDates = substartDate.clone();
+    availableDates.push(Object({
+      startDate:substartDate.format('YYYY-MM-DD HH:MM:SS'),
+      closeDate:newDates.add(keepAliveFor,keepAliveForInterval).format('YYYY-MM-DD HH:MM:SS'),
+      graceDate: allowLateTerm && allowLateInterval ? newDates.add(allowLateTerm,allowLateInterval).format('YYYY-MM-DD HH:MM:SS') : null
+    }));
+  }
+  return availableDates;
+}
+
+
+
+/**
+ * @function getCalculatedCloseSubmissionDate
+ * Get calculated Close date for a Form schedule setting with the given scenario
+ *
+ * @param {Object[]} openedDate An object of Moment JS date
+ * @param {Integer} keepOpenForTerm keepOpenForTerm An integer of number of Days/Weeks OR Years, that tells form to be open for a particular period
+ * @param {String} keepOpenForInterval keepOpenForInterval A string of days,Weeks,months, that tells form to be open for a particular period
+ * @param {Integer} term An integer of number of Days/Weeks OR Years
+ * @param {String} interval A string of days,Weeks,months
+ * @param {Integer} allowLateTerm An integer of number of Days/Weeks OR Years
```

Comment:
What does the number mean? Is it a day, week, or year? How would you specify it? What is this parameter used for?